server/neptune/gateway/event/pull_request_handler.go

```diff
@@ -105,13 +115,44 @@ func (p *ModifiedPullHandler) handle(ctx context.Context, request *http.Buffered
 	if err := p.LegacyHandler.Handle(ctx, request, event, rootCfgs, legacyModeRoots); err != nil {
 		p.Logger.ErrorContext(ctx, err.Error())
 	}
+
 	if err := p.handlePlatformMode(ctx, event, platformModeRoots); err != nil {
 		return errors.Wrap(err, "handling platform mode")
 	}
 	return nil
 }
 
-func (p *ModifiedPullHandler) handlePlatformMode(_ context.Context, _ PullRequest, _ []*valid.MergedProjectCfg) error {
-	// TODO: handle platform mode
+func (p *ModifiedPullHandler) handlePlatformMode(ctx context.Context, event PullRequest, roots []*valid.MergedProjectCfg) error {
+	// skip signaling workflow if no roots
+	if len(roots) == 0 {
+		return nil
+	}
+	// TODO: remove when we begin in-depth testing and rollout of pr mode
+	// feature allocator is only temporary while we continue building out implementation
+	shouldAllocate, err := p.Allocator.ShouldAllocate(feature.PRMode, feature.FeatureContext{
+		RepoName: event.Pull.HeadRepo.FullName,
+	})
+	if err != nil {
+		p.Logger.ErrorContext(ctx, "unable to allocate pr mode")
+		return nil
+	}
+	if !shouldAllocate {
+		p.Logger.InfoContext(ctx, "handler not configured for allocation")
+		return nil
+	}
```

Comment:
We won't leave the allocator here for long. It's only while the initial implementation of PR mode is still in development. Eventually for rollout we will send events to both legacy and temporal workers and instead wrap any mutation behaviors with the allocator.